lib/engine/ability/base.rb

```diff
@@ -9,14 +9,17 @@ class Base
       include Helper::Type
       include Ownable
 
+      attr_accessor :name, :value
       attr_reader :type, :owner_type, :remove, :when, :count
 
-      def initialize(type:, owner_type: nil, count: nil, remove: nil, **opts)
+      def initialize(type:, name: nil, owner_type: nil, count: nil, remove: nil, value: nil, **opts)
         @type = type&.to_sym
+        @name = name.nil? ? @type : name&.to_s
         @owner_type = owner_type&.to_sym
         @when = opts.delete(:when)&.to_s
         @count = count
         @used = false
+        @value = value.nil? ? '' : value&.to_s
```

Comment:
what's the point of value? i don't really see it being used

Reply:
It is used in render_abilities in asserts/app/view/game.corporation.
So e.g. is is used to show the hex name where Coal Field token is assigned, it is used to save the city name for the historical objective.